deploy/notary.go

```diff
@@ -831,6 +831,8 @@ func makeUnsignedDesignateCommitteeNotaryTx(roleContract *rolemgmt.Contract, com
 		return nil, err
 	}
 
+	tx = tx.Copy()
```

Comment:
enough to copy and modify `Signers` if `!tx.Signers[0].Account.Equals(sharedTxData.sender)` only

Reply:
this is kinda hotfix to me, i do not agree with the code here, so something different can be done here in the future. do you really need one more `if` here?

Reply:
btw, this func is called only if `!sharedTxDataMatches` so suggested `if` is already done

Reply:
>  i do not agree with the code here, so something different can be done here in the future

then issue or to draft

Reply:
> this is kinda hotfix to me

nvm, we dont modify other ref type fields, so copy only `Signers`